src/lib/viz/source/SQLSource.ts

```diff
@@ -226,11 +227,14 @@ export class SQLSource extends Source {
 
   private extractMetadataFrom(mapInstance: MapInstance) {
     const { stats } = mapInstance.metadata.layers[0].meta;
-    const geometryType = parseGeometryType(stats.geometryType);
-    const fieldStats = this.getCompleteFieldStats(stats);
+    let geometryType: GeometryType | undefined;
 
-    const metadata = { geometryType, stats: fieldStats };
+    if (stats.geometryType) {
+      geometryType = parseGeometryType(stats.geometryType);
+    }
```

Comment:
Should we at least add a console.warn about empty source? Or maybe even have a source.isEmpty property based on this value?